proxy/proxy.go

```diff
@@ -1540,15 +1542,6 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx.initialSpan = span
 	ctx.parentSpan = span
 
-	defer func() {
-		if ctx.response != nil && ctx.response.Body != nil {
-			err := ctx.response.Body.Close()
-			if err != nil {
-				ctx.Logger().Errorf("error during closing the response body: %v", err)
-			}
-		}
-	}()
```

Comment:
Decided not to pile-up defers but to use straightforward cleanup sequence after streaming the response.
Would like to refactor other defers as well separately.